components/automate-ui/src/app/pages/project/details/project-details.component.ts

```diff
@@ -118,28 +128,26 @@ export class ProjectDetailsComponent implements OnDestroy {
     this.deleteModalVisible = false;
   }
 
-  startRuleDelete(r: any): void {
+  startRuleDelete(rule: Rule): void {
     this.deleteModalVisible = true;
-    this.ruleToDelete = r;
+    this.ruleToDelete = rule;
   }
 
   deleteRule(): void {
-    this.store.dispatch(new DeleteRule({id: this.ruleToDelete.id}));
+    this.store.dispatch(new DeleteRule({ id: this.ruleToDelete.id }));
     this.closeDeleteModal();
   }
 
-  getEditStatus(status: string): string {
-    return status === 'staging'
-      ? 'Edits pending'
-      : 'Applied';
+  getEditStatus(rule: Rule): string {
+    return rule.status === RuleStatus.Staged ? 'Edits pending' : 'Applied';
   }
 
-  showDeleteRule(rule: Rule): boolean {
-    return rule.edits !== 'staging';
+  showDeleteRule(): boolean {
+    return true; // TODO: return false when *project* status is "updating..."
   }
 
   showProjectLink(): boolean {
-    return _find(this.rules, ['edits', 'staging']) ? true : false;
+    return find(['status', RuleStatus.Staged], this.rules) ? true : false;
```

Comment:
If we don't care about _what is returned_, let's use [`some`](https://gist.github.com/jfmengels/6b973b69c491375117dc#_somepredicate-collection) instead:
```suggestion
    return some(['status', RuleStatus.Staged], this.rules);
```

Reply:
Nice!

Reply:
`some` has the added benefit of returning a boolean, too!

Reply:
yup that was my point actually 😉